.github/workflows/e2e-test-linux-vm.yaml

```diff
@@ -5,47 +5,73 @@ on:
-    runs-on: [ubuntu-latest]
+  default-setup:
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Add env variables
-        run: |
-            TAG=$(echo "0.0.$GITHUB_SHA")
-            echo "TAG=$TAG" >> $GITHUB_ENV
-            echo "IMAGE=$IMAGE_REPO:$TAG" >> $GITHUB_ENV
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+
+      - uses: ./.github/actions/setup-arc-e2e
+        id: setup
         with:
-          version: latest
-      - name: Docker Build Test Image
+          github-app-id: ${{secrets.ACTIONS_ACCESS_APP_ID}}
+          github-app-pk: ${{secrets.ACTIONS_ACCESS_PK}}
+          github-app-org: ${{env.TARGET_ORG}}
+          docker-image-name: ${{env.IMAGE_NAME}}
+          docker-image-tag: ${{env.IMAGE_VERSION}}
+
+      - name: Install gha-runner-scale-set-controller
+        id: install_arc_controller
         run: |
-            DOCKER_CLI_EXPERIMENTAL=enabled DOCKER_BUILDKIT=1 docker buildx build --build-arg RUNNER_VERSION=$RUNNER_VERSION --build-arg TAG=$TAG -t $IMAGE . --load
-      - name: Create Kind cluster
+          helm install arc \
+          --namespace "arc-systems" \
+          --create-namespace \
+          --set image.repository=${{ env.IMAGE_NAME }} \
+          --set image.tag=${{ env.IMAGE_VERSION }} \
+          ./charts/gha-runner-scale-set-controller \
+          --debug
+          sleep 30
+          kubectl get pod -n arc-systems
+          kubectl describe deployment arc-gha-runner-scale-set-controller -n arc-systems
+
+      - name: Install gha-runner-scale-set
+        id: install_arc
         run: |
-          PATH=$(go env GOPATH)/bin:$PATH
-          kind create cluster --name $CLUSTER_NAME
-      - name: Load Image to Kind Cluster
-        run: kind load docker-image $IMAGE --name $CLUSTER_NAME
-      - name: Get Token
-        id: get_workflow_token
-        uses: peter-murray/workflow-application-token-action@8e1ba3bf1619726336414f1014e37f17fbadf1db
-        with:
-          application_id: ${{ secrets.ACTIONS_ACCESS_APP_ID }}
-          application_private_key: ${{ secrets.ACTIONS_ACCESS_PK }}
-          organization: ${{ env.TARGET_ORG }}
-      - uses: ./.github/actions/e2e-arc-test
-        with:
-          github-token: ${{ steps.get_workflow_token.outputs.token }}
-          config-url: "https://github.com/actions-runner-controller/arc_e2e_test_dummy"
-          docker-image-repo: $IMAGE_REPO
-          docker-image-tag: $TAG
+          ARC_NAME=arc-runner-${{github.job}}-$(date +'%M-%S')-$(($RANDOM % 100 + 1))
+          helm install "$ARC_NAME" \
+          --namespace "arc-runners" \
+          --create-namespace \
+          --set githubConfigUrl="${{ env.TEST_CONFIG_URL }}" \
+          --set githubConfigSecret.github_token="${{ steps.setup.outputs.token }}" \
+          ./charts/gha-runner-scale-set \
+          --debug
+          echo "ARC_NAME=$ARC_NAME" >> $GITHUB_OUTPUT
+          sleep 30
```

Comment:
Can we maybe use something like `kubectl wait --for-condition=ready pod -n arc-systems {a label to get the controller}`?

Reply:
`kubectl wait` won't work for creating a new resource, it will error if the resource hasn't been fully created.
I will add a short 5 seconds of sleep to give the cluster a little moment.